pkg/service/service.go

```diff
@@ -1775,16 +1780,19 @@ func (s *Service) restoreServicesLocked(svcBackendsById map[lb.BackendID]struct{
 func (s *Service) deleteServiceLocked(svc *svcInfo) error {
 	ipv6 := svc.frontend.L3n4Addr.IsIPv6() || svc.svcNatPolicy == lb.SVCNatPolicyNat46
 	obsoleteBackendIDs := s.deleteBackendsFromCacheLocked(svc)
+	headless := svc.svcType == lb.SVCTypeHeadless
 	scopedLog := log.WithFields(logrus.Fields{
 		logfields.ServiceID: svc.frontend.ID,
 		logfields.ServiceIP: svc.frontend.L3n4Addr,
 		logfields.Backends:  svc.backends,
 	})
 	scopedLog.Debug("Deleting service")
 
-	if err := s.lbmap.DeleteService(svc.frontend, len(svc.backends),
-		svc.useMaglev(), svc.svcNatPolicy); err != nil {
-		return err
+	if !headless {
+		if err := s.lbmap.DeleteService(svc.frontend, len(svc.backends),
+			svc.useMaglev(), svc.svcNatPolicy); err != nil {
+			return err
+		}
 	}
 
 	// Delete affinity matches
```

Comment:
Didn't you want to move the following lines in the `if !headless` case as it was before? :thinking: